src/util/performancetimer.h

```diff
@@ -1,75 +1,38 @@
-/****************************************************************************
-**
-** Copyright (C) 2012 Nokia Corporation and/or its subsidiary(-ies).
-** All rights reserved.
-** Contact: Nokia Corporation (qt-info@nokia.com)
-**
-** This file is part of the QtDeclarative module of the Qt Toolkit.
-**
-** $QT_BEGIN_LICENSE:LGPL$
-** GNU Lesser General Public License Usage
-** This file may be used under the terms of the GNU Lesser General Public
-** License version 2.1 as published by the Free Software Foundation and
-** appearing in the file LICENSE.LGPL included in the packaging of this
-** file. Please review the following information to ensure the GNU Lesser
-** General Public License version 2.1 requirements will be met:
-** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
-**
-** In addition, as a special exception, Nokia gives you certain additional
-** rights. These rights are described in the Nokia Qt LGPL Exception
-** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
-**
-** GNU General Public License Usage
-** Alternatively, this file may be used under the terms of the GNU General
-** Public License version 3.0 as published by the Free Software Foundation
-** and appearing in the file LICENSE.GPL included in the packaging of this
-** file. Please review the following information to ensure the GNU General
-** Public License version 3.0 requirements will be met:
-** http://www.gnu.org/copyleft/gpl.html.
-**
-** Other Usage
-** Alternatively, this file may be used in accordance with the terms and
-** conditions contained in a signed written agreement between you and Nokia.
-**
-**
-**
-**
-**
-** $QT_END_LICENSE$
-**
-****************************************************************************/
-
 #pragma once
+#include <chrono>
 
-//
-// This is a fork of QPerformanceTimer just without the Q prefix
-// To fix interface changes issues in different QT versions
-// Added restart() function.
```

Comment:
Have you checked if this class is still necessary? Maybe we can use plain `QPerformanceTimer` nowadays.

Reply:
QPerformanceTimer doesn't exist anymore as far as I can tell. Closest thing would be QElapsedTimer. Didn't investigate whether that satisfies the same constraints. I will check.

Reply:
QPerformanceTimer  was forked from QElapsedTimer and later reintegrated: 
https://github.com/qt/qt/commit/88966515993773e4d7d6276bce6fccfcc13ab6f4
I think we can adopt that commit to our code and drop our PerformanceTimer entirely. 
I recommend that over the std::chrono::steady_clock solution because it is only weak defined by the C++ standard and may change by any update or different standard library.

Reply:
Unfortunately the API differs a little bit because qt has no duration type. `PerformanceTimer` is also spread over more than 40 classes in our codebase. IMO we should keep PerformanceTimer as a facade over QElapsedTimer that makes use of `mixxx::Duration` (which we should also replace with std::chrono::duration` IMO but thats even harder than replacing PerformanceTimer verbatim).

Reply:
Should we add a task to our Semantic Type Refactoring project https://github.com/mixxxdj/mixxx/projects/5 ?

Reply:
Maybe. We should at least think about replacing the `mixxx::Duration` internals with `std::chrono::duration` to allow for gradual transition. But `mixxx::Duration` is already a semantic type, so I don't see a pressing need. The only major advantage `std::chrono::duration` has is its `duration_cast` as far as I can tell.